### ISSUE TITLE
Document insert / update fails with simple List of Strings as value

### ISSUE THREAD

**Issue by sfxcode** [MongoCamp/mongocamp-server#45]
Example:

```json
{"name":"test1","number":1234,"children":["test","test1"]}
```

fails with Error Message for Error 500:

```
class java.lang.String cannot be cast to class scala.collection.immutable.Map (java.lang.String is in module java.base of loader 'bootstrap'; scala.collection.immutable.Map is in unnamed module of loader 'app')
```

**Comment by sfxcode** [comment]
Using Array of objects is ok

```json
{"name":"test1","number":1234,"children":[{"name": "test"}]} 
```
works ...

**Comment by QuadStingray** [comment]
type erasure in pattern matching fixed in version 1.4.2
